test/TestUtilities/ApproxArray.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef SFML_TESTUTILITIES_APPROXARRAY_HPP
+#define SFML_TESTUTILITIES_APPROXARRAY_HPP
+
+#include <doctest.h>
+#include <array>
+
+namespace std
+{
+    template <typename T, std::size_t size>
+    std::ostream& operator<<(std::ostream& out, const std::array<T, size>& array)
+    {
+        if (size == 0)
+            return out << "{}";
+
+        out << "{ ";
+        for (size_t i = 0; i + 1 < array.size(); ++i)
+            out << array[i] << ", ";
+        out << array.back() << " }";
```

Comment:
We may need to increase the precision of the `ostream` to make sure all significant information is printed otherwise you may issues where something like `1.00000 == 1.00000` gets printed to console when the real numbers are something like `1.000000000 == 1.000000211`. I had this issue when writing the `sf::Angle` tests and later realized I needed the following to make sure the full precision of a `float` got printed.
```
out << std::fixed << std::setprecision(std::numeric_limits<float>::max_digits10);
```

Reply:
I would not have thought of it. Thank you. I don't think `std::fixed` is needed though.
The precision is now appropriately set for arrays of floating-point values.